src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableHttpClientCRLCheck.cs

```diff
@@ -0,0 +1,213 @@
+    class DoNotDisableHttpClientCRLCheck : DiagnosticAnalyzer
+    {
+        internal static DiagnosticDescriptor DefinitelyDisableHttpClientCRLCheckRule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5399",
+            nameof(MicrosoftNetCoreAnalyzersResources.DefinitelyDisableHttpClientCRLCheck),
+            nameof(MicrosoftNetCoreAnalyzersResources.DefinitelyDisableHttpClientCRLCheckMessage),
+            DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            helpLinkUri: null,
+            descriptionResourceStringName: nameof(MicrosoftNetCoreAnalyzersResources.DoNotDisableHttpClientCRLCheckDescription),
+            customTags: WellKnownDiagnosticTagsExtensions.DataflowAndTelemetry);
+        internal static DiagnosticDescriptor MaybeDisableHttpClientCRLCheckRule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5400",
+            nameof(MicrosoftNetCoreAnalyzersResources.MaybeDisableHttpClientCRLCheck),
+            nameof(MicrosoftNetCoreAnalyzersResources.MaybeDisableHttpClientCRLCheckMessage),
+            DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            helpLinkUri: null,
+            descriptionResourceStringName: nameof(MicrosoftNetCoreAnalyzersResources.DoNotDisableHttpClientCRLCheckDescription),
+            customTags: WellKnownDiagnosticTagsExtensions.DataflowAndTelemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(
+                DefinitelyDisableHttpClientCRLCheckRule,
+                MaybeDisableHttpClientCRLCheckRule);
+
+        /// <summary>
+        /// PropertySetAbstractValue index for CheckCertificateRevocationList property.
+        /// </summary>
+        private const int CheckCertificateRevocationListIndex = 0;
+
+        private static readonly ConstructorMapper ConstructorMapper = new ConstructorMapper(
+            (IMethodSymbol constructorMethod, IReadOnlyList<PointsToAbstractValue> argumentPointsToAbstractValues) =>
+            {
+                return PropertySetAbstractValue.GetInstance(PropertySetAbstractValueKind.Flagged);
+            });
+
+        private static readonly PropertyMapperCollection PropertyMappers = new PropertyMapperCollection(
+            new PropertyMapper(
+                "CheckCertificateRevocationList",
+                (ValueContentAbstractValue valueContentAbstractValue) =>
+                    PropertySetCallbacks.EvaluateLiteralValues(
+                        valueContentAbstractValue,
+                        (object o) => o is bool booleanValue && booleanValue == false),
+                CheckCertificateRevocationListIndex));
+
+        private static readonly HazardousUsageEvaluatorCollection HazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+            new HazardousUsageEvaluator(
+                WellKnownTypeNames.SystemNetHttpHttpClient,
+                ".ctor",
+                "handler",
```

Comment:
>handler [](start = 17, length = 7)

If we do handle derived classes, the parameter name might not be the same as HttpClient's constructor's.

Reply:
Yes. For these specific cases, people can add a new HazardousUsageEvaluator for that derived classes.